guide/C/ch_currency.xml

```diff
@@ -34,793 +35,1132 @@
+    <guilabel>Use Trading Accounts</guilabel> checkbox.</para>
+
+    <sect2 id="trading_acct_concepts">
+    <title>Trading Account Concepts</title>
+
+    <para>In accounting terms, trading accounts are specialized equity accounts
+      to record changes in revenue which result from transactions between two or more
+      currencies. In &app; they are implemented as a top level account
+      type of Trading. When trading accounts are
+      enabled the required sub accounts of Trading are automatically created if
+      they do not already exist when a transaction between one or more currencies
+      is created.
+    </para>
+    <para>In the extended Accounting Equation, Trading accounts would appear as
+      follows:
+      <equation>
+        <mathphrase>Assets = Liabilities + Equity (+ Income -Expenses) + Trading
+        </mathphrase>
+      </equation>
+    </para>
+
+    <caution><para> The user is prevented from manually making entries directly
+      to the registers for any of the accounts in the Trading top level account
+      group. All entries to these accounts are automatically generated when
+      transactions between currencies are created in other account registers
+      within the &app;account heirarchy.</para>
+    </caution>
+
+    <para>The Trading account group is structured below the top level account
+    <guilabel>Trading</guilabel> with a sub-account <guilabel>CURRENCY</guilabel>
+      which in turn has a sub-account for each currency between which
+      transactions have been made. For the example previously used in
+      <link linkend="currency_manual">Manually Recording Currency Transactions</link>,
+      the additional accounts which will be created automatically
+      when transactions are created between different currencies
+      with trading accounts enabled are:</para>
+<literallayout><emphasis role="bold">
+        Trading
+            CURRENCY
+                EUR
+                HKD
+                USD</emphasis>
+</literallayout>
+      <caution><para>The names of these accounts are again assigned automatically and
+        cannot be changed by a user without disrupting the functionality
+        of the trading accounts in &app;.</para>
+      </caution>
+    </sect2>
+    <sect2 id="currency_trading">
+      <title>Recording Foreign Currency Transactions with Trading Accounts</title>
+      <sect3 id="currency_trading_account_setup">
+        <title>Setup of Accounts for Trading Accounts</title>
+        <para>The same account structure useds in the manual recording of
+          currency transactions in the previous section is required. The
+          initial account configuration for Assets and Equity accounts are
+          as follows:</para>
+        <screenshot id="currency__trading_acct_setup">
+          <mediaobject>
+            <imageobject role="html">
+              <imagedata fileref="figures/currency_trading_acct_setup.png"
```

Comment:
2 Issues:
* The important part is not displayed: The trading accounts. You should resize the window before capturing the screenshot again.
* This is IMHO already wrong in 12.3:
Equity and its subaccounts like `Opening Balance` should be only in your local currency.
We should not give the users bad examples.
Or missed I a reference In the later parts of this chapter? I suspect the original author tried to explain: As long as there is no exchange rate, the value `Total (<local currency>)` of foreign currencies is displayed as 0. We should try to find an better example to demonstrate it in the common part.

Then the price would not entered manually, but by the exchange dialog while creating the opening balance transactions.
NB.: Also the exchange dialog should usually "Pieces" in both currencies instead of an inaccurate exchange rate.

Reply:
> Equity and its subaccounts like Opening Balance should be only in your local currency.

Then you'd only be able to record income and expense in local currency so the exchanges wouldn't be modeled in the books, hiding currency exchange gains and losses.

Reply:
Frank,

The trading accounts did not exist at that point as they are not created until you add a transaction between currencies.

Re 12.3  There is an assumtion here that a user does not have foreign currency accounts in place with existing balances when a book is created or for those who start a new book for each new year for example. Is that a necessary restriction to place on GnuCash?

 I could remove any reference to how the accounts reached there initial state and perhaps make the example a bit more realistic for someone actually involved in currency trading. I don't think most would regard an 0.82c loss with $60.00 in costs a viable business transaction. Having explored how trading accounts work I can probably come up with a more realistic example now.

Reply:
This was really a first pass attempt constructed while exploring trading accounts to better understand them myself. The treatment in the 12.2 part could probably be revised in the light of Selinger's article to use manual recording using the trading account concept or perhaps dropped completely as Frank suggested. 

I haven't yet checked out what happens for example when a currency price change occurs and is recorded in the price database independent of a currency transaction between the currencies. This would affect the conversion to the Report currency in the Accounts tab but what effect it has on reports for example i.e. how do trading losses and gains in the Balance sheet when they are unrealized gains as opposed to the realized gains here. I flagged it as something to explore further while doing this exercise.

I followed the original author in setting up the accounts. How the accounts are setup is irrelevant to the operation of the trading accounts themselves!

Reply:
> > Equity and its subaccounts like Opening Balance should be only in your local currency.
> 
> Then you'd only be able to record income and expense in local currency so the exchanges wouldn't be modeled in the books, hiding currency exchange gains and losses.

Hm, `local currency` is ambiguous. I meant in the book currency. For the german rules you could talk with CPA assistant "Frakturfreak" on IRC.

Reply:
Frank,
See the problem. Under the Handelgesetzbuch, gains and losses from currency transactions are recognized in profit and loss as part of other operating Income or Other Operating Expenses in an income statement rather than as line items in the Balance Sheet. 

however, the IFRS also treats gains and losses from foreign currency transactions as part of profit and loss as does the FASB in the US AFAIK ( only a brief scan of the FASB rules) and the AASB  in Australia. Germany however does not recognize "comprehensive income" which can include unrealized gains and losses as well as realized gains and losses (depends primarily on whether they want to tax you on them or not - usually not as it is against basic tax principles) in some jurisdictions.  I haven't yet ascertained how trading accounts in GnuCash are affected by unrealized gains (to be investigated) and whether the reported gains and losses are totally realized or a combination of realized and unrealized losses.

 Normally the transactions to the trading accounts occur at the exchange rate current at the time of the transactions. What is presented in the Accounts tab in the Report currency column will be affected by any changes to the currency's exchange rate recorded in the price database, but Reports in most jurisdictions for foreign currencies seem to be based on the exchange rates at the report date

I must admit I was somewhat confused that trading gains and losses were reported in the Balance Sheet in the reports.  I think a lot of these issues relate more to presentation in reports than the underlying recording in GnuCash but correctly translating what is recorded in GnuCash to the reports.

David

Reply:
> I must admit I was somewhat confused that trading gains and losses were reported in the Balance Sheet in the reports. I think a lot of these issues relate more to presentation in reports than the underlying recording in GnuCash but correctly translating what is recorded in GnuCash to the reports.

It falls out from the use of pricedb prices (either nearest in time or latest) to calculate the value of the other-commodity accounts in the report currency. Since those values are different from the transaction values it creates an imbalance that must be accounted for somewhere. The report author chose (reasonably, IMO) to create an "unrealized gains" line for it.

Reply:
It seems, to fulfill the 

- **low value** principle (traditional HGB), german and probably others users must use a _cost based_ price source (with the downside no longer having a foot note with the pending result).

- **fair value** principle (IFRS) users have to choose one of the _time based_ sources.

This still needs to be incorporated in tooltips and docs - after verification.